posthog/queries/util.py

```diff
@@ -50,15 +51,18 @@ def format_ch_timestamp(timestamp: datetime, convert_to_timezone: Optional[str]
 
 
 def get_earliest_timestamp(team_id: int) -> datetime:
-    results = insight_sync_execute(
-        GET_EARLIEST_TIMESTAMP_SQL,
-        {"team_id": team_id, "earliest_timestamp": EARLIEST_TIMESTAMP},
-        query_type="get_earliest_timestamp",
-    )
-    if len(results) > 0:
-        return results[0][0]
-    else:
-        return timezone.now() - DEFAULT_EARLIEST_TIME_DELTA
+    def query_for_earliest_timestamp():
```

Comment:
I think we should be caching only positive responses here and not cache cases where no events exist.

Reply:
To elaborate:
- This is for not blocking things for onboarding
- If we wanted to get fancy, accept team as an argument to this function, and only cache if a positive response _and_ team is e.g. older than a day.

Reply:
Note - I'm willing to let this slide if you don't want to get into this.

Reply:
Hmmm, I'm being slow :) how does it block onboarding?